src/main/java/com/github/checkstyle/regression/git/DiffParser.java

```diff
@@ -59,24 +65,28 @@ private DiffParser() {
      */
     public static List<GitChange> parse(String repositoryPath, String branchName)
             throws IOException, GitAPIException {
-        final List<GitChange> returnValue;
+        final List<GitChange> returnValue = new LinkedList<>();
         final File gitDir = new File(repositoryPath, ".git");
         final Repository repository = new FileRepositoryBuilder().setGitDir(gitDir)
                 .readEnvironment().findGitDir().build();
 
         try {
             final TreeParserPair pair = getTreeParserPair(repository, branchName);
             final Git git = new Git(repository);
+            final DiffFormatter formatter = new DiffFormatter(DisabledOutputStream.INSTANCE);
+            formatter.setRepository(repository);
 
             try {
-                returnValue = git.diff()
+                final List<DiffEntry> diffs = git.diff()
                         .setOldTree(pair.commonAncestorTreeParser)
                         .setNewTree(pair.prTreeParser)
                         .call()
                         .stream()
                         .filter(entry -> entry.getChangeType() != DiffEntry.ChangeType.DELETE)
-                        .map(DiffParser::convertDiffEntryToGitChange)
```

Comment:
Yes. I mentioned this at the post:
> we have to use for-loop instead of map, for convertDiffEntryToGitChange now throws IOException

There can't be checked exception in lambda